flink-tests/src/test/java/org/apache/flink/test/checkpointing/IgnoreInFlightDataITCase.java

```diff
@@ -117,12 +118,13 @@ private boolean executeIgnoreInFlightDataDuringRecovery() {
         env.enableCheckpointing(10);
         env.disableOperatorChaining();
         env.getCheckpointConfig().enableUnalignedCheckpoints();
+        env.getCheckpointConfig().setAlignmentTimeout(Duration.ZERO);
         env.getCheckpointConfig().setCheckpointingMode(CheckpointingMode.EXACTLY_ONCE);
         env.getCheckpointConfig().setCheckpointIdOfIgnoredInFlightData(1);
         env.setRestartStrategy(fixedDelayRestart(1, 0));
 
         env.addSource(new NumberSource(lastCheckpointValue))
-                .shuffle()
+                .rebalance()
```

Comment:
Why is this necessary?

Reply:
This test should guarantee that all three subtasks will receive at least one record. Right now, the source emits fifty records which are enough to make this test working even with `shuffle` but strictly to say `shuffle` doesn't make sense here while `rebalance` does exactly what we need. So this change is not really necessary but it provides more correct behaviour for this test.

Reply:
Why can't we use the default `forward` partitioner then?

Reply:
Perhaps, I don't understand something but who guarantees that all subtasks will receive the records? As I can see in the implementation of ForwardPartitioner it always returns the first subpartition(`return 0;`)

Reply:
Does your source have different parallelism than the following map operator? I don't think so. Using `ForwardPartitioner` will result in subtask 0 of the source pushing to subtask 0 of the map operator and so on. If your only goal is to have each of the map operators receive data, I think that's enough, isn't it? 

Rebalance means that each source subtask will produce to every map subtask.

Reply:
> Does your source have lower parallelism than the following map operator? 
Actually, it does. I have only one source and three followed maps.I did it because it is easier to manage the execution when you have only one source.

Reply:
Are you sure? Where do you set the parallelism to `1` for the source? I can't seem to find it.

Reply:
Yes, I am sure. It happens because I implemented SourceFunction which always has parallelism 1 rather than ParallelSourceFunction which has the configured parallelism.

Reply:
Right, I keep forgetting about the `ParallelSourceFuntion` :facepalm: It's such a design failure imo.

Reply:
Still, I believe you don't need to specify the partitioning explicitly. If you remove the `rebalance` whatsoever it will use the `RescalePartitioner` which behaves the same as `RebalancePartitioner` in this case, because you have just a single source task.

If you do believe a `rebalance` is better, add a comment why you need the `rebalance` there. It's not straightforward to figure out the role of it.

Reply:
Ok, I see it works pretty well even without `rebalance`